### PR TITLE
Make sure the import tests exit their hosts

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] ppy/osu-framework#1374

Since apparently this is the way we should be going forward...

### CODE REVIEW

**Comment by smoogipoo** [comment]
@peppy you can't possibly think this is a good idea, I don't believe it.

File: osu.Game.Tests/Beatmaps/IO/ImportBeatmapTest.cs [hunk 1]
```diff
@@ -64,6 +66,9 @@ public void TestImportOverIPC()
                 ensureLoaded(osu);
 
                 waitForOrAssert(() => !File.Exists(temp), "Temporary still exists after IPC import", 5000);
+
+                host.Exit();
+                client.Exit();
```
Comment (peppy) [inline osu.Game.Tests/Beatmaps/IO/ImportBeatmapTest.cs]: as discussed, unnecessary. apart from this i do believe this is how we want to do things (as long as `host.Run` is a blocking operation)

**Comment by smoogipoo** [comment]
@Tom94 I need your input on this, because @peppy and I can't reach an agreement here.

I'm adamant that `Dispose` should call Exit and should stop all threads, rather than relying on the user to do it themselves. This is a pattern I find with basically every single other .NET Framework object - filestreams, webrequests, etc... E.g. calling Dispose() on a FileStream will call Close() on it, calling Dispose() an a StreamReader will call Dispose() on the underlying stream and that will in turn call Close() on itself.

**Comment by peppy** [comment]
Going by this logic, `host.Run` would need to throw an `ObjectDisposed` or `OperationCanceled` exception on being disposed by an external force.

File: osu.Game.Tests/Beatmaps/IO/ImportBeatmapTest.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Tests/Beatmaps/IO/ImportBeatmapTest.cs]: the `client.Exit()` can still be removed here, right?

**Review by peppy — APPROVED** [review]

**Pull request merged**